capellambse/helpers.py

```diff
@@ -68,6 +74,11 @@ def remove_whitespace(text: str):
         yield remove_whitespace(element.tail)
 
 
+def is_uuid_string(string: t.Any) -> te.TypeGuard[UUIDString]:
+    """Validate that ``string`` is a valid UUID."""
+    return isinstance(string, str) and bool(RE_VALID_UUID.fullmatch(string))
```

Comment:
Reading your commits is a gold mine for me to grow. The official [PEP 647](https://peps.python.org/pep-0647/) explains what `TypeGuards` do and how to use them. There is also a way to use them for validation of `TypedDict`s. Awesome!